supertokens_python/recipe/multitenancy/recipe_implementation.py

```diff
@@ -54,46 +56,205 @@ def __init__(self, querier: Querier, config: MultitenancyConfig):
     async def get_tenant_id(
         self, tenant_id_from_frontend: Optional[str], user_context: Dict[str, Any]
     ) -> Optional[str]:
-        pass
+        return tenant_id_from_frontend
 
     async def create_or_update_tenant(
         self,
         tenant_id: Optional[str],
         config: TenantConfig,
         user_context: Dict[str, Any],
     ) -> CreateOrUpdateTenantOkResult:
-        raise NotImplementedError
+        response = await self.querier.send_put_request(
+            NormalisedURLPath("/recipe/multitenancy/tenant"),
+            {
+                "tenantId": tenant_id,
+                **config.to_json(),
+            },
+        )
+        return CreateOrUpdateTenantOkResult(
+            created_new=response["createdNew"],
+        )
 
     async def delete_tenant(
         self, tenant_id: str, user_context: Dict[str, Any]
     ) -> DeleteTenantOkResult:
-        raise NotImplementedError
+        response = await self.querier.send_post_request(
+            NormalisedURLPath("/recipe/multitenancy/tenant/remove"),
+            {"tenantId": tenant_id},
+        )
+        return DeleteTenantOkResult(
+            did_exist=response["didExist"],
+        )
 
     async def get_tenant(
         self, tenant_id: Optional[str], user_context: Dict[str, Any]
     ) -> GetTenantOkResult:
-        _ = self.querier.send_get_request(
+        from supertokens_python.recipe.thirdparty.provider import (
+            ProviderConfig,
+            UserInfoMap,
+            UserFields,
+            ProviderClientConfig,
+        )
+
+        res = await self.querier.send_get_request(
             NormalisedURLPath(
                 f"{tenant_id or DEFAULT_TENANT_ID}/recipe/multitenancy/tenant"
             ),
-            {},
         )
 
-        # FIXME: Fill values from the response
-        # if "status" in response and response["status"] == "OK":
-        #     pass
+        providers: List[ProviderConfig] = []
```

Comment:
how is this tested?

Reply:
Just added tests for this in `test_tenant_thirdparty_config` of `test_tenants_crud.py`.